sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -137,13 +137,21 @@ object RewritePredicateSubquery extends Rule[LogicalPlan] with PredicateHelper {
       plan: LogicalPlan): (Option[Expression], LogicalPlan) = {
     var newPlan = plan
     val newExprs = exprs.map { e =>
-      e transformUp {
+      e transformDown {
```

Comment:
to be able to see Not(In) first before (In) ?

Reply:
@mgaido91 
>> I don't see any test (please correct me if I am wrong) where multiple attributes are used as output of the subquery. Can we add and compare with other RDBMS? Thanks.

In [here](https://github.com/apache/spark/blob/844a3ff82a688e7398bb130a44750aec78420698/sql/core/src/test/resources/sql-tests/inputs/subquery/in-subquery/nested-not-in.sql#L113-L134) ? Is this what you meant ?

Reply:
yes, thanks, but that doesn't test when the outer values are null, right? I think it would be good to have also cases with:
 - more than 2 attributes;
 - with the outer values being null;
 - complex data types involved (eg. structs)

What do you think?

Reply:
@mgaido91 How can i say "no" to more testing :-) ? I will add the tests. Thanks !!

Reply:
thanks @dilipbiswal :)